providers/amazon/src/airflow/providers/amazon/aws/transfers/http_to_s3.py

```diff
@@ -164,12 +167,21 @@ def s3_hook(self) -> S3Hook:
     def execute(self, context: Context):
         self.log.info("Calling HTTP method")
         response = self.http_hook.run(self.endpoint, self.data, self.headers, self.extra_options)
-
-        self.s3_hook.load_bytes(
-            response.content,
-            self.s3_key,
-            self.s3_bucket,
-            self.replace,
-            self.encrypt,
-            self.acl_policy,
-        )
+        if self.extra_options.get("stream", False):
+            self.s3_hook.load_file_obj(
+                response.raw,
+                self.s3_key,
+                self.s3_bucket,
+                self.replace,
+                self.encrypt,
+                self.acl_policy,
+            )
+        else:
+            self.s3_hook.load_bytes(
+                response.content,
+                self.s3_key,
+                self.s3_bucket,
+                self.replace,
+                self.encrypt,
+                self.acl_policy,
+            )
```

Comment:
what about something like

```suggestion
        common_args = (
            self.s3_key,
            self.s3_bucket,
            self.replace,
            self.encrypt,
            self.acl_policy,
        )
        if self.extra_options.get("stream", False):
            self.s3_hook.load_file_obj(response.raw, *common_args)
        else:
            self.s3_hook.load_bytes(response.content, *common_args)
```